### DISCUSSION TITLE
Issue when passing nested structure

### DISCUSSION THREAD

**Discussion by mstodulski** [webonyx/graphql-php#1558] [Q&A]
Hi,
I've been using your library for a while and it's working great, but I hit a problem. When attempting to execute query with passing args to create an object that contains another object, which contains a collection of another objects, I receive a weird error message:

`Field "arrayOfComplexObjectsLevel3" is not defined by type "complexObjectLevel2". Did you mean "arrayOfComplexObjectsLevel3"?`

Problem can be reproduced using the code of service placed here:

https://github.com/mstodulski/graphql-example

and quering the service using curl from the command line:

`curl http://localhost -H 'content-type: application/json' -d '{"query": "mutation { addComplexObject( name: \"Example complex object name\", complexObjectLevel2: { arrayOfComplexObjectsLevel3: [ { nameLevel3: \"Example level 3 name 1\" }, { nameLevel3: \"Example level 3 name 2\" }, { nameLevel3: \"Example level 3 name 3\" } ] } ) { name } }"}'`

Could you help me with this? It seems to me this structure should be accepted according to the GraphQL standard.

**Comment by spawnia** [comment]
I guess the issue is that you are using a mismatched key for the field: https://github.com/mstodulski/graphql-example/blob/e4ac5ae27699c10d23011fc854b66da221662a73/src/Struct/MutationType.php#L55

I strongly advise not hand-rolling your server implementation using this library directly. Consider using a framework such as https://github.com/nuwave/lighthouse.

**Comment by mstodulski** [comment]
Hi,
I apologize sincerely. I've refactored a larger project into a simpler, easier-to-analyze form, and something went wrong...
Anyway, i've already fixed the key of this field, and the result is still the same.
Could you help me with this?

**Comment by mstodulski** [comment]
I also noticed that if we change this line of code:
https://github.com/mstodulski/graphql-example/blob/5e44e13df882b851fdf77b31de6de338c0a76bbd/src/Struct/MutationType.php#L57
`'type' => Type::listOf(Types::complexObjectLevel3()),`
to this one:
`'type' => Type::listOf(Type::int()),`
then the validator no longer reports an issue with the absence of the arrayOfComplexObjectsLevel3 field.
That's why i suspect there's some issue with DocumentValidator::validate() function or arguments provided to it.
